src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,61 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static final int INITIAL_SIZE = 10;
+    private static final double GROWTH_FACTOR = 1.5;
+    private int currentSize = INITIAL_SIZE;
+    private int size = 0;
+    private T[] table = createArray(currentSize);
 
+    private T[] createArray(int size) {
+        return (T[]) new Object[size];
+    }
+
+    public void push(T value) {
+        table = grow();
+        table[size] = value;
+        size++;
     }
 
     public T peek() {
-        return null;
+        if (size == 0) {
+            throw new EmptyStackException();
+        }
```

Comment:
You are already calling the exact same check 2 times in such a small class - move this logic to a separate private method.

Reply:
ok, done